sdk/cargo-build-sbf/src/main.rs

```diff
@@ -1048,6 +1066,13 @@ fn main() {
                 .takes_value(false)
                 .conflicts_with("force_tools_install")
                 .help("Skip downloading and installing platform-tools, assuming they are properly mounted"),
+            )
+            .arg(
+                Arg::new("no_rustup")
+                .long("no-rustup")
+                .takes_value(false)
+                .conflicts_with("force_tools_install")
+                .help("Do not use rustup to manage the toolchain"),
```

Comment:
I'm worried this argument will confuse some people, and they won't understand how to use it necessarily. Can we add some more info about why it's needed and how to use it? I took an attempt at it
```suggestion
                .help("Do not use rustup to manage the toolchain. By default, cargo-build-sbf invokes rustup to find the Solana rustc using a `+solana` toolchain override. This flag disables that behavior."),
```